prody/proteins/blastpdb.py

```diff
@@ -1,50 +1,65 @@
 # -*- coding: utf-8 -*-
-"""This module defines functions for blast searching Protein Data Bank."""
+"""This module defines functions for blast searching the Protein Data Bank."""
 
 import os.path
-import os
 
-from prody import LOGGER
+from prody import LOGGER, PY3K
 from prody.utilities import dictElement, openURL, which
+from prody.sequence.msafile import parseMSA
 
-__all__ = ['PDBBlastRecord', 'blastPDB', 'showSequenceTree']
+import platform, os, re, sys, time, urllib
+
+if PY3K:
+    import urllib.parse as urllib
+    import urllib.request as urllib2
+else:
+    import urllib
+    import urllib2
+
+from prody.sequence import Sequence
+from prody.atomic import Atomic
+from prody.proteins.pdbfile import parsePDB
 
+__all__ = ['PDBBlastRecord', 'blastPDB', 'showSequenceTree']
 
-def blastPDB(sequence, filename=None, **kwargs):
+def blastPDB(sequence='runexample', filename=None, **kwargs):
     """Returns a :class:`PDBBlastRecord` instance that contains results from
-    blast searching of ProteinDataBank database *sequence* using NCBI blastp.
+    blast searching *sequence* against the PDB using NCBI blastp.
 
-    :arg sequence: single-letter code amino acid sequence of the protein
-        without any gap characters, all white spaces will be removed
-    :type sequence: str
+    :arg sequence: an object with an associated sequence string 
+         or a sequence string itself
+    :type sequence: :class:`Atomic`, :class:`Sequence`, or str
 
     :arg filename: a *filename* to save the results in XML format
     :type filename: str
 
     *hitlist_size* (default is ``250``) and *expect* (default is ``1e-10``)
     search parameters can be adjusted by the user.  *sleep* keyword argument
     (default is ``2`` seconds) determines how long to wait to reconnect for
-    results.  Sleep time is doubled when results are not ready.  *timeout*
-    (default is 120s) determines when to give up waiting for the results.
+    results.  Sleep time is multiplied by 1.5 when results are not ready.  
+    *timeout* (default is 120 s) determines when to give up waiting for the results.
     """
 
     if sequence == 'runexample':
         sequence = ('ASFPVEILPFLYLGCAKDSTNLDVLEEFGIKYILNVTPNLPNLFENAGEFKYKQIPI'
                     'SDHWSQNLSQFFPEAISFIDEARGKNCGVLVHSLAGISRSVTVTVAYLMQKLNLSMN'
                     'DAYDIVKMKKSNISPNFNFMGQLLDFERTL')
+    elif isinstance(sequence, Atomic):
+        sequence = sequence.select('calpha and chain %s' % chain).getSequence()
+    elif isinstance(sequence, Sequence):
+        sequence = str(sequence)
+    elif isinstance(sequence, str):
+        if len(sequence) == 4 or len(sequence) == 5:
+            ag = parsePDB(sequence)
+            sequence = ag.select('calpha and chain %s' % chain).getSequence()
     else:
-        try:
-            sequence = ''.join(sequence.split())
-            _ = sequence.isalpha()
```

Comment:
I think this interesting piece of code is to remove spaces, tabs, line breakers from the sequence. This could be useful when the sequence is long because those characters are invisible and they can be easily left in the string when copying & pasting. Please consider adding this preprocessing.

Reply:
I restored the middle line. The rest of checking and I assumed this was too.